### DISCUSSION TITLE
how to set a constant pressure case?

### DISCUSSION THREAD

**Discussion by czc-zju** [AMReX-Combustion/PeleLMeX#132] [Q&A]
we want to set the pressure as one atm during all the simulation process, so how to set the perturbational pressure($\Pi$) as 0?
![3fe0b3eac7ada1a7bc5aa6bbb25a816](https://user-images.githubusercontent.com/81547457/195973486-e7bd7f07-38bc-4ac2-bf9a-388a96c6d66a.png)

**Comment by drummerdoc** [comment]
The low Mach model says that the total pressure is spatially constant to order Mach number squared. However pressure gradients are still required to drive the flow. When you declare the perturbational pressure is identically zero there is no flow. I doubt that’s what you want :) 

If when running you increase the number of SDC iterations, that will drive the thermodynamic pressure toward being spatially constant without requiring that the perturbational pressure vanish. Set up the code to do a single time step several times with increasing values of the number of sdc iterations and look at the RhoRT variable in the plotfile. Use this to decide how many sdc iterations you need for your problem, but make sure you monitor this as the time increases away from the initial steps when dt is (optionally) scaled down with init_shrink, and the reactions aren’t yet releasing heat, etc. For larger stiffer chemistry models we have found that you may need to increase the number of sdc iterations to reach an acceptably constant RhoRT field 


Also it is important to note that there are other schemes that enforce constant RhoRT by using that with continuity to determine T, thus ignoring energy conservation. This approach is very popular because it’s easy to understand. However it leads to horrible robustness issues, requiring excessive grid resolution near density gradients. Even for those codes, it is the perturbational pressure gradients that are responsible for advective flows.
